zip-0320.rst

```diff
@@ -157,33 +157,32 @@ address into a Unified Address [#zip-0316-unified-addresses]_.
 Motivations for Alternative 2
 -----------------------------
 
-Traceable Unified Addresses fit into the existing Zcash Unified Address
-ecosystem. Existing Consumers that support Unified Addresses will automatically
-be able to recognize a Traceable Unified Address as a valid Zcash address, but
-will not be able to send to that address unless they update their code to
-understand the new Receiver Typecode defined in this ZIP. Even in the case that
-Traceable P2PKH Receivers are not understood by the sending wallet, a Unified
-Address-supporting wallet will be able to automatically provide good error
-messages for their users to indicate that the wallet needs to be updated to
-understand and send to these addresses.
-
-In addition, by integrating with the Unified Address framework, it becomes
-possible for the addresses being generated to include extra metadata; in
-particular, metadata items such as an Address Expiry Height or Address Expiry
-Date [#zip-0316-address-expiry]_ may be included. For exchange use cases such
-as Binance's, it is useful to ensure that an address provided to a user has a
+Traceable Unified Addresses fit into the Zcash Unified Address ecosystem
+defined by ZIP 316, Revision 1 [#zip-0316-revision-1]_. Existing Consumers of
+Unified Addresses will not be able to send to these address unless they update
+their code to understand the new MUST-understand Metadata Typecode defined in
+this ZIP. 
+
+By integrating with the Unified Address framework, it becomes possible for the
+addresses being generated to include extra metadata; in particular, metadata
+items such as an Address Expiry Height or Address Expiry Date
+[#zip-0316-address-expiry]_ may be included. For exchange use cases such as
+Binance's, it is useful to ensure that an address provided to a user has a
 limited utility life, such that after expiration the user must obtain a new
 address in order to be able to continue to send funds
 [#binance-address-expiry]_.
 
 Specification (Alternative 2)
 -----------------------------
 
-Upon activation of this ZIP, the section `Encoding of Unified Addresses` of ZIP
-316 [#zip-0316-unified-addresses]_ will be modified to define a new Traceable
-P2PKH Receiver Type having Typecode :math:`\mathtt{0x04}`, the value of which
-MUST be the 20-byte **validating key hash** of a Zcash P2PKH Address as defined
-in [#protocol-transparentaddrencoding]_.
+Upon activation of this ZIP, the section `Metadata Items` of ZIP 316
+[#zip-0316-metadata-items]_ will be modified to define a new MUST-understand
+Metadata Item type: Source Restriction Metadata, having Typecode
+:math:`\mathtt{0xE2}`, the value of which must be a single byte:
```

Comment:
Should it always be a single byte, or should it just be constrained currently to be a single byte if that first byte is 0x00? I guess it doesn't matter which way it is specified now, as parsers need to understand both the typecode and its value, so a different-length value is no different to a different-valued single-byte value.

Reply:
:heavy_check_mark: